src/bit-systems/waypoint.ts

```diff
@@ -164,6 +171,16 @@ export function waypointSystem(
       }
     }
 
+    const hashUpdated = window.location.hash !== "" && previousWaypointHash !== window.location.hash;
+    if (hashUpdated && initialSpawnHappened) {
+      const waypointName = window.location.hash.replace("#", "");
+      if (eid) {
+        moveToWaypoint(world, eid, characterController, previousWaypointHash === null);
+        window.history.replaceState(null, "", window.location.href.split("#")[0]); // Reset so you can re-activate the same waypoint
+      }
+      previousWaypointHash = window.location.hash;
+    }
```

Comment:
Can you elaborate why `initialSpawnHapped` is needed for me for review?

Reply:
Looking at the code and this issue https://github.com/mozilla/hubs/issues/2833 I believe is so you can spawn to a specific named waypoint and the scene default spawn point does not override it when entering.

Reply:
Would you mind adding an inline comment about it? It is not straightforward to new readers.

Reply:
I think the comment need more explanation because it may be unclear to new readers about how `initialSpawnHappened` can avoid the override. May be you can refer https://github.com/mozilla/hubs/pull/2837/files#r468103137

And again, I don't think `initialSpawnHappened` and `previousWaypointHash` are good approaches in systems because they sound like state in systems to me. It's ideal if we can come up with proper approaches. Or we may think of managing them in components. At lease we may need comment that they need to be improved later.

Reply:
Added the PR comment from above to the comment.